python/tests/integration/test_retry.py

```diff
@@ -0,0 +1,216 @@
+# Copyright (c) 2015 Uber Technologies, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+from __future__ import absolute_import
+
+import pytest
+import tornado
+import tornado.gen
+from mock import patch
+
+from tchannel.errors import ProtocolError
+from tchannel.errors import TChannelError
+from tchannel.errors import TimeoutError
+from tchannel.messages import ErrorCode
+from tchannel.transport_header import RetryType
+from tchannel.tornado import Request
+from tchannel.tornado import TChannel
+from tchannel.tornado.peer import should_retry_on_error
+from tchannel.tornado.stream import InMemStream
+
+
+@pytest.mark.gen_test(timeout=10)
+def test_retry_timeout():
+    endpoint = b'tchannelretrytest'
+
+    @tornado.gen.coroutine
+    def handler_error(request, response, proxy):
+        yield tornado.gen.sleep(2)
```

Comment:
It won'd actually test multiply seconds. Here just needs a number that is greater than the timeout_per_request. So the client will timeout.